test/jdk/javax/swing/JFrame/bug4419914.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2001, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4419914
+ * @summary Tests that tab movement is correct in RTL component orientation.
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual bug4419914
+*/
+
+import java.awt.BorderLayout;
+import java.awt.ComponentOrientation;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import java.util.Locale;
+
+public class bug4419914 {
+    private static final String INSTRUCTIONS = """
+        1. You will see a frame with five buttons.
+        2. Confirm that each button is placed as follows:
+             NORTH
+        END  CENTER  START
+             SOUTH
+        3. Press the "NORTH" button and confirm the button is focused.
+        4. Press TAB repeatedly and confirm that the TAB focus moves from right to left.
+             (NORTH - START - CENTER - END - SOUTH - NORTH - START - CENTER - ...)
+
+            If there's anything different from the above items, click Fail else click Pass.""";
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("Tab movement Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(12)
+                .columns(42)
+                .testUI(bug4419914::createTestUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static JFrame createTestUI() {
+        JFrame frame = new JFrame("bug4419914");
+        frame.setFocusCycleRoot(true);
+        frame.setComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+        frame.setLocale(Locale.ENGLISH);
+
+        frame.enableInputMethods(false);
+        frame.getContentPane().setComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+        frame.getContentPane().setLocale(Locale.ENGLISH);
```

Comment:
Thanks for the information, didn't know that..Updated....there is around ~400 occurrence of getContentPane() in swing tests in open still so probably something to be optimised in future...

Reply:
The tests were probably written before Java 5, even tests written later could still use `getContentPane` so that the test could be run with previous versions of Java. Yet it's irrelevant now.

The Java Tutorials have never been updated, [How to Make Frames](https://docs.oracle.com/javase/tutorial/uiswing/components/frame.html) uses `getContentPane` explicitly.

Reply:
Good point..this backward support still keep us using obsolete things..
Java tutorials also there are few bugs like [6417303](https://bugs.openjdk.org/browse/JDK-6417303) which needs fix in the tutorial..dont know whom to ask or can we do it ourselves..

Reply:
You're right. Cleaning up the code makes it easier to read.